genie-web/src/main/java/com/netflix/genie/web/properties/AgentFilterProperties.java

```diff
@@ -0,0 +1,98 @@
+/*
+ *
+ *  Copyright 2019 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.web.properties;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.EnvironmentAware;
+import org.springframework.core.env.Environment;
+import org.springframework.validation.annotation.Validated;
+
+import javax.annotation.Nullable;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Properties for the {@link com.netflix.genie.web.services.AgentFilterService}.
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+@ConfigurationProperties
```

Comment:
I'd recommend putting the `PROPERTY_PREFIX` in here in the `prefix` field

Reply:
I had to remove it, it would fail to bind.
That's because this class does not actually bind any property during startup.
The ENABLE property key is defined here, but used to enable the configuration.
And the rest of these properties are dynamic.